### PR TITLE
Adding units tests for Lookup3Hash, MurmurHash, BloomCalculations and Varint

### PR DESCRIPTION
**Pull request by paulkennethkent**
The test cases contained in this PR were written by Diffblue's Deeptest software. These tests are intended to help you detect regressions caused by future code changes.

One of our missions is to provide useful tests to open source projects. In return, we would appreciate your feedback on our tests to help us further improve our tool.

Results for various open source projects can be viewed at:

http://diff.blue/demo

### CODE REVIEW

**Comment by paulkennethkent** [comment]
I will clean up the branch names and add licences after the review.

**Review by JohnLBergqvist — APPROVED** [review]
Remember to add license headers.

File: src/test/java/com/clearspring/analytics/hash/Lookup3HashTest.java [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+    final int length = -1_610_612_735;
+    final int initval = -2_147_483_648;
+
+    // Act
+    final int retval = Lookup3Hash.lookup3ycs(k, offset, length, initval);
+
+    // Assert result
+    Assert.assertEquals(1_588_444_911, retval);
+  }
+}
```
Comment (sudo-jaa) [inline src/test/java/com/clearspring/analytics/hash/Lookup3HashTest.java:144]: These all look good!

**Review by sudo-jaa — APPROVED** [review]

File: src/test/java/com/clearspring/analytics/stream/membership/BloomCalculationsTest.java [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+    final BloomSpecification retval = BloomCalculations.computeBucketsAndK(maxFalsePosProb);
+
+    // Assert result
+    Assert.assertNotNull(retval);
+    Assert.assertEquals(3, retval.bucketsPerElement);
+    Assert.assertEquals(1, retval.K);
+  }
+
+  @Test
+  public void constructorInputZeroZeroOutputVoid() {
```
Comment (petersear) [inline src/test/java/com/clearspring/analytics/stream/membership/BloomCalculationsTest.java]: Do we want this void constructor test?

File: src/test/java/com/clearspring/analytics/stream/membership/BloomCalculationsTest.java [hunk 3]
```diff
@@ -0,0 +1,137 @@
...
+    final int bucketsPerElement = 0;
+
+    // Act, creating object to test constructor
+    BloomSpecification objectUnderTest = new BloomSpecification(k, bucketsPerElement);
+
+    // Method returns void, testing that no exception is thrown
+  }
+
+  @Test
+  public void constructorOutputVoid() {
```
Comment (petersear) [inline src/test/java/com/clearspring/analytics/stream/membership/BloomCalculationsTest.java]: Another void constructor test

**Review by petersear — APPROVED** [review]

**Pull request closed without merging**